types/scheduler.d.ts

```diff
@@ -0,0 +1,142 @@
+interface TaskPriorityChangeEventInit extends EventInit {
+  /** A string indicating the previous priority of the task. One of `"user-blocking"`, `"user-visible"`, or `"background"`. */
+  previousPriority: TaskPriority;
+}
+
+declare global {
+  /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Scheduler) */
+  interface Scheduler {
+    /**
+     * Adds a task to the scheduler as a callback, optionally specifying a priority, delay, and/or a signal for aborting the task.
+     * @param callback An callback function that implements the task. The return value of the callback is used to resolve the promise returned by this function.
+     * @param options {@link SchedulerPostTaskOptions} options.
+     *
+     * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Scheduler/postTask)
+     */
+    postTask<T extends unknown>(callback: () => T, options?: SchedulerPostTaskOptions): Promise<T>;
+    /**
+     * Returns a promise that yields to the event loop when awaited. Optionally specify a priority and/or a signal for aborting the task.
+     * @param options {@link SchedulerYieldOptions} options.
+     */
+    yield(options?: SchedulerYieldOptions): Promise<void>;
+  }
+
+  /**
+   * A controller object that can be used to both abort and change the priority of one or more prioritized tasks.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TaskController)
+   */
+  class TaskController extends AbortController {
+    constructor(options?: TaskControllerOptions);
+    /** Returns a {@link TaskSignal} instance. The signal is passed to tasks so that they can be aborted or re-prioritized by the controller. */
+    readonly signal: TaskSignal;
+    /**
+     * Sets the priority of the controller's signal, and hence the priority of any tasks with which it is associated.
+     *
+     * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TaskController/setPriority)
+     */
+    setPriority(priority: TaskPriority): void;
+  }
+
+  /**
+   * The `prioritychange` event, sent to a {@link TaskSignal} if its priority is changed.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TaskPriorityChangeEvent)
+   */
+  class TaskPriorityChangeEvent extends Event {
+    constructor(type: string, init: TaskPriorityChangeEventInit);
+    /**
+     * The priority of the corresponding {@link TaskSignal} _before_ this prioritychange event.
+     *
+     * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TaskPriorityChangeEvent/previousPriority)
+     */
+    readonly previousPriority: TaskPriority;
+  }
+
+  /**
+   * A signal object that allows you to communicate with a prioritized task, aborting it or changing the priority (if required) via a {@link TaskController} object.
+   *
+   * [MDN Reference](https://developer.mozilla.org/docs/Web/API/TaskSignal)
+   */
+  class TaskSignal extends AbortSignal {
```

Comment:
I don't know if this matters, but FYI we couldn't subclass AbortSignal for the polyfill -- instead we had to define properties on the underlying `AbortSignal`. But if this just treats a `TaskController`'s signal as a `TaskSignal`, matching the properties we added, this is great.

Reply:
> I don't know if this matters, but FYI we couldn't subclass AbortSignal for the polyfill

Right, I saw that. For anyone using the typescript compiler, at least, I believe it shouldn't matter because it's an AbortSignal with a `priority` and an `onprioritychange`, which is exactly what this type is checked as (class names and the prototype hierarchy generally aren't typed or checked)